src/errors.rs

```diff
@@ -27,6 +27,7 @@ pub enum BcryptError {
     InvalidBase64(base64::DecodeError),
     #[cfg(any(feature = "alloc", feature = "std"))]
     Rand(getrandom::Error),
+    Truncation(usize),
```

Comment:
This should have some docs that it's only used if the `error_on_truncation` feature is enabled.